### PR TITLE
fix(RegularGrid): removed deprecated functions, cleared multiple defi…

### PR DESCRIPTION
**Pull request by MelchiorSchuh**
…nitions, clarified grid accessors names.

### CODE REVIEW

File: include/geode/mesh/core/regular_grid.h [hunk 1]
```diff
@@ -101,59 +98,65 @@ namespace geode
 
         const Point< dimension >& origin() const;
 
+        AttributeManager& cell_attribute_manager() const;
```
Comment (BotellaA) [inline include/geode/mesh/core/regular_grid.h:101]: Why moving them? Most of the classes attribute managers are at the bottom. But why not.

File: src/geode/mesh/core/regular_grid.cpp [hunk 2]
```diff
@@ -346,17 +338,12 @@ namespace geode
...
-        }
-
-        absl::optional< GridCellIndices< dimension > > cell(
+        absl::optional< GridCellsAroundVertex< dimension > > cell(
             const Point< dimension >& query ) const
         {
-            GridCellIndices< dimension > indices;
-            GridCellIndex< dimension > min;
-            GridCellIndex< dimension > max;
+            GridCellsAroundVertex< dimension > cells_around_point;
```
Comment (BotellaA) [inline src/geode/mesh/core/regular_grid.cpp:344]: To declare only at line 370?

**Review by BotellaA — APPROVED** [review]

File: include/geode/mesh/core/regular_grid.h (hunk 1 above)
Comment (MelchiorSchuh) [reply include/geode/mesh/core/regular_grid.h:101]: I just felt that theses accessors to important features should be visible before the functions computing some stuff on the class.

**Review by panquez — APPROVED** [review]

**Pull request merged**